### ISSUE TITLE
gradients using loss=F.make_loss(myloss) are not same as gradients not using loss=F.make_loss(myloss) in Hybrid programming

### ISSUE THREAD

**Issue by Sundrops** [apache/incubator-mxnet#19657]
> This operator accepts a customized loss function symbol as a terminal loss and the symbol should be an operator with no backward dependency. The output of this function is the gradient of loss with respect to the input data.

The description of ndarray.make_loss is same as the description of symbol.make_loss. And it only explains symbol, not ndarray. 
I want to know what `F.make_ loss()` will do when I use `net.hybridize()` and `loss.backward()`.

https://mxnet.apache.org/versions/1.7.0/api/python/docs/api/ndarray/ndarray.html?highlight=make_loss#mxnet.ndarray.make_loss
https://mxnet.apache.org/versions/1.7.0/api/python/docs/api/symbol/symbol.html#mxnet.symbol.make_loss

```python
class Myloss(mx.gluon.nn.HybridBlock):
    def __init__(self):
        super(Myloss, self).__init__()
    def hybrid_forward(self, F, pred, gt):
        loss_l2 = F.sum(F.square(pred - gt), axis=1) / 2
        return F.make_loss(loss_l2)
net=resnet()
net.hybridize()
x = net(input)
myloss = Myloss()
loss = myloss(x, y)
loss.backward()
```

**Comment by szha** [comment]
@Sundrops there's no need to use make_loss in Gluon as all values can be used as head gradients now.
